pulsar-discovery-service/src/test/java/org/apache/pulsar/discovery/service/web/DiscoveryServiceWebTest.java

```diff
@@ -78,10 +80,14 @@
  */
 public class DiscoveryServiceWebTest extends BaseZKStarterTest{
 
+    private static final Logger log = LoggerFactory.getLogger(DiscoveryServiceWebTest.class);
+
     private Client client = ClientBuilder.newClient(new ClientConfig().register(LoggingFeature.class));
     private static final String TLS_SERVER_CERT_FILE_PATH = "./src/test/resources/certificate/server.crt";
     private static final String TLS_SERVER_KEY_FILE_PATH = "./src/test/resources/certificate/server.key";
-
+    // DiscoveryServiceServlet gets initialized by a server and this map will help to retrieve ZK while mocking
+    // DiscoveryServiceServlet
+    private static final Map<String, MetadataStoreExtended> metadataStoreInstanceCache = Maps.newConcurrentMap();
```

Comment:
it looks like we are not clearing this map at the end of the test.
we will leak MetadataStoreExtended instances

Reply:
It's not leaking. static map is intentional because of the `DiscoveryServiceServlet` which gets initialized by Webserver and we need to inject MetadataStore into dynamically initialized servelet. so, `DiscoveryServiceServletTest`  extends the discovery-servlet class and access metadata store from this static map.
It's not leaking because everytime when test starts it puts the metadatastore into the map and then it closes the metadata store at the end of test so, it will not leak. 
So, static map is the way to access metadata-store from dynamically initialized servlet which we can't mock or control.

Reply:
I am fine with having the static map. it is a good idea.

I would only add a 
```
@AlfterClass
public void clearMetadataStoreInstanceCache() { 
    metadataStoreInstanceCache.clear();
}
```

if we do not clear the map then we are keeping a reference to the (closed) MetadataStore into a static variable.